### PR TITLE
Support `metric_names` introduced in Optuna

### PR DESCRIPTION
**Pull request by c-bata**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

Refs https://github.com/optuna/optuna/pull/4383

## What does this implement/fix? Explain your changes.

`optuna_dashboard.set_objective_names` is back-ported to Optuna's repository by @HideakiImamura. This PR deprecates the `set_objective_names()` function and uses `metric_names` if included in `system_attrs`.

### CODE REVIEW

File: optuna_dashboard/_named_objectives.py [hunk 1]
```diff
@@ -22,6 +27,17 @@ def set_objective_names(study: optuna.Study, names: list[str]) -> None:
           study = optuna.create_study(directions=["minimize", "minimize"])
           set_objective_names(study, ["val_loss", "flops"])
     """
+
+    if hasattr(study, "set_metric_names"):
+        # TODO(c-bata): List this warning on errors.rst in docs.
```
Comment (c-bata) [inline optuna_dashboard/_named_objectives.py:32]: I will address this todo item after #427 is merged since `errors.rst` will be added in it.

**Comment by c-bata** [comment]
@HideakiImamura @knshnb Could you review this PR if you have time?

File: optuna_dashboard/_named_objectives.py (hunk 1 above)
Comment (knshnb) [reply optuna_dashboard/_named_objectives.py:32]: Can we address this in this PR because #427 will be merged soon?

Comment (c-bata) [reply optuna_dashboard/_named_objectives.py:32]: Yes, that makes sense 👍 I will address this in this PR.

**Comment by c-bata** [comment]
I've pushed a commit to update the `errors.html` in our documentation. PTAL.

Screenshot 2023-04-17 18 41 24

**Review by knshnb — APPROVED** [review]
Sorry for the delayed review. LGTM!

**Pull request merged**